src/main/java/org/skypro/skyshop/controller/ShopControllerAdvice.java

```diff
@@ -0,0 +1,18 @@
+package org.skypro.skyshop.controller;
+
+import org.skypro.skyshop.model.exceptions.NoSuchProductException;
+import org.skypro.skyshop.model.exceptions.ShopError;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
```

Comment:
```suggestion
@RestControllerAdvice
```
для RestController-ов надо использовать RestControllerAdvice аннотацию